### PR TITLE
32 modeler developer preview

### LINKED ISSUE #32: Add Web Modeler to Docker Compose
this issue is for tracking only. The original issue is this: https://github.com/camunda/web-modeler/issues/2204

### PR DESCRIPTION
**Pull request by pihme**
closes #32

### CODE REVIEW

**Comment by pihme** [comment]
> What is the tarball used for?

Will be removed during cleanup

**Comment by pihme** [comment]
@jfriedenstab Please have a look, if there is still time before your holidays. There is still one open TODO - to replace the latest version with the actual release tag. Other than that it is complete.

I kept the ports now as is. Didn't want to meddle with it too much so short before the deadline.

**Comment by pihme** [comment]
There is one known issue with this setup: https://github.com/camunda/web-modeler/issues/2609

File: docker-compose-web-modeler-beta.yaml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      PUSHER_APP_KEY: modeler-app-key 
+      PUSHER_APP_SECRET: modeler-app-secret
+      PUSHER_APP_CLUSTER: modeler
+    networks:
+      - modeler      
+
+  mailhog:
+    # If you want to use your own SMTP server, you can remove this container
+    # and configure RESTAPI_MAIL_HOST, RESTAPI_MAIL_PORT and RESTAPI_MAIL_ENABLE_TLS
+    # accordingly
```
Comment (jfriedenstab) [inline docker-compose-web-modeler-beta.yaml:62]: Real SMTP servers probably require authentication:

```suggestion
    # If you want to use your own SMTP server, you can remove this container
    # and configure RESTAPI_MAIL_HOST, RESTAPI_MAIL_PORT, RESTAPI_MAIL_USER, RESTAPI_MAIL_PASSWORD, and RESTAPI_MAIL_ENABLE_TLS
    # accordingly
```

File: docker-compose-web-modeler-beta.yaml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      RESTAPI_PUSHER_HOST: modeler-websockets
+      RESTAPI_PUSHER_PORT: "8060"
+      RESTAPI_PUSHER_APP_ID: modeler-app
+      RESTAPI_PUSHER_KEY: modeler-app-key
+      RESTAPI_PUSHER_SECRET: modeler-app-secret
+      RESTAPI_OAUTH2_TOKEN_AUDIENCE: web-modeler-api
+      RESTAPI_OAUTH2_TOKEN_ISSUER: http://localhost:18080/auth/realms/camunda-platform
+      RESTAPI_OAUTH2_TOKEN_ISSUER_BACKEND_URL: http://keycloak:8080/auth/realms/camunda-platform
+      RESTAPI_IDENTITY_BASE_URL: http://identity:8084/          
+      RESTAPI_SERVER_URL: http://localhost:8081
```
Comment (jfriedenstab) [inline docker-compose-web-modeler-beta.yaml:107]: `RESTAPI_SERVER_URL` has to use the exposed port of the webapp: `http://localhost:8070`
It is used, for example, to construct the URL for project invitations (which are opened in the browser).

File: docker-compose-web-modeler-beta.yaml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      retries: 30
+    environment:
+      POSTGRES_DB: modeler-db
+      POSTGRES_USER: modeler-db-user
+      POSTGRES_PASSWORD: modeler-db-password
+    networks:
+      - modeler      
+
+  modeler-websockets:
+    container_name: websockets
```
Comment (jfriedenstab) [inline docker-compose-web-modeler-beta.yaml:42]:
```suggestion
    container_name: modeler-websockets
```
(The other containers' names are all prefixed)

**Review by jfriedenstab — CHANGES_REQUESTED** [review]
Nice! Just a couple of smaller remarks.

**Comment by jfriedenstab** [comment]
One more thing that might be noteworthy in the `README` file:
When you want to deploy a diagram to the local Zeebe cluster you need to use the name of the Zeebe container as the host name instead of `localhost` (so the cluster endpoint would be `zeebe:26500`). This is different from the Desktop Modeler.

**Comment by pihme** [comment]
@christian-konrad Please have a look at the README.md changes in particular

**Review by jfriedenstab — APPROVED** [review]
:tada:

**Review by christian-konrad — APPROVED** [review]
I like it! Well done

**Pull request merged**